src/test/java/com/iexec/common/utils/ContextualLockRunnerTests.java

```diff
@@ -156,9 +152,20 @@ void getWithLockOnParity() {
     private <K> void run(BiConsumer<Integer, Map<K, Integer>> action) {
         final Map<K, Integer> remainingCallsPerKey = new ConcurrentHashMap<>();
 
-        IntStream.range(0, NUMBER_OF_THREADS)
-                .parallel()
-                .forEach(threadPosition -> action.accept(threadPosition, remainingCallsPerKey));
+        try {
+            IntStream.range(0, NUMBER_OF_THREADS)
+                    .parallel()
+                    .forEach(threadPosition -> action.accept(threadPosition, remainingCallsPerKey));
+        } catch (AssertionError e) {
+            // Sometimes it has a cause, sometimes not.
+            // When there's a cause,
+            // the "Synchronization failed" message is in this cause.
+            // Otherwise, it's in the exception itself.
+            if (e.getCause() != null) {
+                throw (AssertionError)e.getCause();
```

Comment:
Are we sure the cause can always be cast to an `AssertionError` ?
As `assertThatThrownBy` works with `Throwable`, it may be safer to `throw e.getCause()`
```suggestion
                throw e.getCause();
```

Reply:
I'm 99% sure it's always an `AssertionError`. If we get anything else, that's another issue and I would prefer the test to fail if there's an unknown issue.

Reply:
You have a poin